spirv_glsl.cpp

```diff
@@ -3396,13 +3440,12 @@ std::string CompilerGLSL::flattened_access_chain_vector_scalar(uint32_t base, co
 
 std::pair<std::string, uint32_t> CompilerGLSL::flattened_access_chain_offset(uint32_t base, const uint32_t *indices,
                                                                              uint32_t count, uint32_t offset,
-                                                                             bool *need_transpose)
+                                                                             bool *need_transpose,
+                                                                             uint32_t *out_matrix_stride)
 {
 	const auto *type = &expression_type(base);
-	uint32_t type_size = 0;
-
-	// For resolving array accesses, etc, keep a local copy for poking.
-	SPIRType temp;
+	uint32_t current_type = type->self;
```

Comment:
Consider renaming to type_id and leaving a comment as to why it's different from type->self